LSPS2/README.md

```diff
@@ -0,0 +1,736 @@
+    which also defines a timeout for the rest of this flow; if the
+    flow does not complete in that timeout, the parameters are no
+    longer valid and the LSP will refuse to open the channel.
+* The client requests for a JIT channel, specifying how much it will
+  receive, via a `lsps2.buy` request.
+* The LSP returns an SCID, which identifies this request to the LSP.
+* The client generates an invoice, which includes the above SCID and the
+  LSP node ID as a routehint.
+* The client hands the invoice to whoever it will receive funds from.
+* The payment is forwarded to the LSP.
+* The LSP recognizes the next hop SCID as being a JIT channel request,
+  and opens a 0-confirmation channel to the client, which must be
+  connected to the LSP at that time.
+* The LSP forwards the payment to the client, deducting the channel
+  opening fee.
+* The client claims the payment.
+
+> **Rationale** An alternative flow would be for the client to provide a
+> payment hash for the LSP to recognize, instead of the LSP providing a
+> SCID for the client to use in its invoice.
+> However, this has drawbacks for some future modifications of the Lightning
+> protocol:
+>
+> - Protocols where the payer generates the payment preimage and gives
+>   the preimage to the payee inside the encrypted onion are incompatible
+>   with having the client (the payee) provide the payment hash to the
+>   LSP beforehand.
+>   - Current plans for asynchronous receive uses such a scheme.
+> - PTLCs are supposed to improve privacy by having each hop have a
+>   different blinding factor, but for the LSP to recognize the payment
+>   point, it would have to know the blinding factor towards the client,
+>   which reduces the privacy of the client in a multipath payment.
+>   - PTLCs would allow a payer to buy a private key from a payee
+>     (client in this context); normally the per-hop blinding factors
+>     would hide the sold private key from forwarding nodes, but due to
+>     the LSP having to know the exact payment point and the blinding
+>     factor towards the client / payee, it would also learn the sold
+>     private key.
+
+### 1. API Information
+
+`lsps2.getinfo` is the entrypoint for each client using the API.
+It indicates supported versions of this protocol, as well as any limits the
+LSP imposes, and parameters for payment.
+
+The client MUST request `lsps2.getinfo` to read:
+
+- the supported `versions` of the LSP.
+- the `opening_fee` of the LSP and its related parameters.
+
+The client MUST then determine protocol compatibility (if it supports a
+`version` that the LSP also supports), and also determine if the fee
+imposed by the LSP is acceptable.
+
+`lsps2.getinfo` takes no parameters `{}`, and has no errors defined.
+
+Example `lsps2.getinfo` result:
+
+```JSON
+{
+    "versions": [1],
+    "opening_fee_params_menu": [
+        {
+            "base": "546000",
+            "proportional": 1200,
```

Comment:
I think if one number is encoded as a string, it's better to encode all numbers as a string for easier serialization/deserialization. You could use a global setting to encode/decode numbers as strings then.

Reply:
`base` is intended to be a monetary amount, while `proportional` is a parts-per-million number.  Monetary amounts need to be strings as monetary amounts in millisatoshis may exceed the JSON-specified IEEE double-precision float range, but parts-per-million are definitely going to fit as it would never exceed 1,000,000.